src/goto-programs/goto_convert_exceptions.cpp

```diff
@@ -128,6 +128,71 @@ void goto_convertt::convert_msc_leave(
 
 /*******************************************************************\
 
+Function: goto_convertt::convert_java_try_catch
+
+  Inputs:
+
+ Outputs:
+
+ Purpose:
+
+\*******************************************************************/
+
+void goto_convertt::convert_java_try_catch(
+  const codet &code,
+  goto_programt &dest)
+{
+  assert(code.operands().size()>=1);
+
+  // add the CATCH instruction to 'dest'
+  goto_programt::targett catch_instruction=dest.add_instruction();
+  catch_instruction->make_catch();
+  catch_instruction->code.set_statement(ID_catch);
+  catch_instruction->source_location=code.source_location();
```

Comment:
this should have its `function` set (check also other instructions you generate)